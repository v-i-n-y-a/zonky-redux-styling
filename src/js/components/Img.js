import React from "react";
import _ from "lodash";

const Img = ({ src, alt }) => {
  if (!src || typeof src !== "string") return <img />;
  const props = { src: `/loans/marketplace${src}`, alt: alt ? alt : "" };
  return <img {...props} class="img-responsive" />;
};
export default Img;
