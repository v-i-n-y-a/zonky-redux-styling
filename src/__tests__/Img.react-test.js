import React from 'react';
import Img from '../js/components/Img.js';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

describe('<Img  />', () => {
  it('Img without any properties', () => {
    const component = renderer.create(<Img />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Img with props should normal img tag', () => {
    const component = renderer
      .create(<Img src="test.jpg" alt="Test Image" />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
