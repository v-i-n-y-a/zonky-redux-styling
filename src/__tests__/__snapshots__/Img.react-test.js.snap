// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Img  /> Img with props should normal img tag 1`] = `
<img
  alt="Test Image"
  class="img-responsive"
  src="/loans/marketplacetest.jpg"
/>
`;

exports[`<Img  /> Img without any properties 1`] = `<img />`;
